time.js

```diff
@@ -1,12 +1,169 @@
-
 require('@js-joda/timezone');
 const time = require('@js-joda/core');
 
 //openHAB uses a RFC DateTime string, js-joda defaults to the ISO version, this defaults RFC instead
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
+
+/**
+ * When formatter is not supplied, uses an RFC DateTimeFormatter that can parse the format
+ * of java.time.ZonedDateTime.toString()
+ * @param {String} text 
+ * @param {time.DateTimeFormatter} formatter 
+ * @returns {time.ZonedDateTime}
+ * @throws {DateTimeParseException} when the text cannot be parsed
+ */
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
     return targetParse(text, formatter);
 }
 
-module.exports = time;
+/**
+ * A passthrough function that can allow users to call `toZonedDateTime` on
+ * almost anything that can be converted to a time.ZonedDateTime without first
+ * checking to see if it's already a time.ZonedDateTime. 
+ * @returns {time.ZonedDateTime} this
+ */
+time.ZonedDateTime.prototype.toZonedDateTime = function () {
+    return this;
+}
+
+/**
+ * Change the date of the passed in dt to match the date. This preserves
+ * the hours and minutes even in cases where dt and date span a DST changeover.
+ * @param {time.ZonedDateTime} dt date time to move the date on
+ * @param {time.ZonedDateTime} date date time to move dt's date to
+ * @returns {time.ZonedDateTime} dt with date's date
+ */
+const changeDate = function (dt, date) {
+    return dt.withYear(date.year())
+             .withMonth(date.month())
+             .withDayOfMonth(date.dayOfMonth());
+}
+
+/**
+ * Moves the date portion of the date time to today, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with today's date
+ */
+time.ZonedDateTime.prototype.toToday = function () {
```

Comment:
`now()` just gives me the current *time* and date. 

This function replaces the date with today's date, leaving then time alone but accounting for DST changes. 

It's particularly useful in cases where, for example, someone has a DateTimeType representing an alarm or something like that that they want to keep using day after day to schedule a Timer or something like that. This function will move the DT to today's date, preserving the time so it can be used today.

Reply:
Gotcha, makes sense, maybe "Moves the date portion of the date time to today, while maintaining the existing time, accounting for DST"  or something,  it also might be worth putting an example in the comment.